json11.hpp

```diff
@@ -106,14 +106,14 @@ class Json final {
     Json(const T & t) : Json(t.to_json()) {}
 
     // Implicit constructor: map-like objects (std::map, std::unordered_map, etc)
-    template <class M, typename std::enable_if<
+    template <class M, typename = decltype(std::declval<M>().begin()), typename std::enable_if<
```

Comment:
Is there a way to work this into the enable_if?  Seems like it becomes a bit harder to understand since you're doing two different forms of compatibility checks.  This one is direct SFINAE while enable_if hides it inside of the more boolean syntax.